File: mainActivity/src/main/java/com/sierrawireless/avphone/task/AlertRuleClient.java
package com.sierrawireless.avphone.task;

import java.io.IOException;

import net.airvantage.model.AirVantageException;
import net.airvantage.model.alert.v1.AlertRule;
import net.airvantage.utils.IAirVantageClient;

import com.sierrawireless.avphone.model.AvPhoneApplication;

public class AlertRuleClient implements IAlertRuleClient {

    private IAirVantageClient client;

    AlertRuleClient(IAirVantageClient client) {
        this.client = client;
    }
    
    @Override
    public AlertRule getAlertRule(String serialNumber, String application) throws IOException, AirVantageException {
        String alertRuleName = AvPhoneApplication.ALERT_RULE_NAME;
        return client.getAlertRuleByName(alertRuleName, application);
    }
   
    @Override
    public void createAlertRule(String application) throws IOException, AirVantageException {
        AlertRule alertRule = AvPhoneApplication.INSTANCE.createAlertRule();
        client.createAlertRule(alertRule, application);
    }

    
}

File: mainActivity/src/main/java/com/sierrawireless/avphone/task/AlertRuleClient.kt
package com.sierrawireless.avphone.task

import java.io.IOException

import net.airvantage.model.AirVantageException
import net.airvantage.model.alert.v1.AlertRule
import net.airvantage.utils.IAirVantageClient

import com.sierrawireless.avphone.model.AvPhoneApplication
import com.sierrawireless.avphone.tools.Tools
import net.airvantage.model.AvSystem

class AlertRuleClient internal constructor(private val client: IAirVantageClient) : IAlertRuleClient {

    @Throws(IOException::class, AirVantageException::class)
    override fun getAlertRule(serialNumber: String, system: AvSystem): AlertRule? {
        return client.getAlertRuleByName(Tools.buildAlertName(), system)
    }

    override fun getAlertState(uid: String): Boolean {
        return client.getAlertState(uid)
    }

    @Throws(IOException::class, AirVantageException::class)
    override fun createAlertRule(Application: String, system: AvSystem, name:String) {
        val alertRule = AvPhoneApplication.createAlertRule(system, name)
        client.createAlertRule(alertRule, Application, system)
    }


    @Throws(IOException::class, AirVantageException::class)
    override fun updateAlertRule(Application: String, system: AvSystem, alertRule: AlertRule, name:String) {
        AvPhoneApplication.updateAlertRule(system, alertRule, name)
        client.updateAlertRule(alertRule, Application, system)
    }

    @Throws(IOException::class, AirVantageException::class)
    override fun deleteAlertRule(alert: AlertRule) {
        client.deleteAlertRule(alert)
    }



}
